thirdparty/CMakeLists.txt

```diff
@@ -3,76 +3,54 @@ include(FetchContent)
 ###################################################################################################
 # - spdlog ----------------------------------------------------------------------------------------
 
-FetchContent_Declare(
-    spdlog
-    GIT_REPOSITORY https://github.com/gabime/spdlog.git
-    GIT_TAG        v1.7.0
-    GIT_SHALLOW    true
-)
-
-FetchContent_GetProperties(spdlog)
-if(NOT spdlog_POPULATED)
-  FetchContent_Populate(spdlog)
-  add_subdirectory(${spdlog_SOURCE_DIR} ${spdlog_BINARY_DIR} EXCLUDE_FROM_ALL)
-endif()
+CPMAddPackage(
+  NAME spdlog
+  GITHUB_REPOSITORY gabime/spdlog
+  VERSION 1.7.0
+  GIT_SHALLOW TRUE
+  )
 
 ###################################################################################################
 # - thrust/cub ------------------------------------------------------------------------------------
 
 message(STATUS "Fetching Thrust")
 
-FetchContent_Declare(
-    thrust
-    GIT_REPOSITORY https://github.com/thrust/thrust.git
-    # August 28, 2020
-    GIT_TAG        52a8bda46c5c2128414d1d47f546b486ff0be2f0
+CPMAddPackage(
+  NAME thrust
+  GITHUB_REPOSITORY thrust/thrust
+  GIT_TAG 1.10.0
+  VERSION 1.10.0
+  DOWNLOAD_ONLY TRUE
```

Comment:
Should this be marked `GIT_SHALLOW TRUE` as well?

Reply:
Yes, good point. Now fixed.